src/main/java/com/google/devtools/build/lib/remote/blobstore/http/HttpDownloadHandler.java

```diff
@@ -95,15 +99,15 @@ protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Ex
 
       ByteBuf content = ((HttpContent) msg).content();
       bytesReceived += content.readableBytes();
-      if (downloadSucceeded) {
-        content.readBytes(out, content.readableBytes());
-      }
+      content.readBytes(out, content.readableBytes());
       if (bytesReceived == contentLength || msg instanceof LastHttpContent) {
         if (downloadSucceeded) {
           succeedAndReset(ctx);
         } else {
-          HttpException error =
-              new HttpException(response, "Download failed with status: " + response.status(), null);
+          String errorMsg = response.status() + "\n";
+          errorMsg += new String(((ByteArrayOutputStream) out).toByteArray());
```

Comment:
Don't you need to specify an encoding, e.g. UTF_8?

Reply:
done